### PR TITLE
[AMDGPU][GFX12] Support disassembling MUBUF instructions with arbitrary FORMAT values.

### PR DESCRIPTION
**Pull request by kosarev**
Some tools generate such instructions with the FORMAT field set to 0, which corresponds to buf_fmt_invalid, but that should not prevent them from being recognised on decoding.

### CODE REVIEW

**Comment by kosarev** [comment]
A more comprehensive solution allowing proper reassembling would give the instructions the format operand, but I'm not sure we want go that far.

File: llvm/lib/Target/AMDGPU/BUFInstructions.td [hunk 1]
```diff
@@ -2447,7 +2447,13 @@ multiclass VBUFFER_MUBUF_Real_gfx12<bits<8> op, string real_name> {
   def _gfx12 : VBUFFER_Real_gfx12<op, ps, real_name> {
     // Set the last bit of format to 1 to avoid round-trip issues, as some tools
     // print BUF_FMT_INVALID for format 0.
-    let Inst{55} = 0b1;
+    let Inst{61-55} = 1;
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/BUFInstructions.td:2450]: I'm not sure how to read this. Is it setting the 6-bit value to 1, as in mostly zeroes? Given the comment, can we keep just setting bit 55 to 1 and then set the others to 0?

Comment (kosarev) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td:2450]: The idea is to give the `_gfx12` version a fully defined FORMAT value and then just let the `_gfx12_format` thing catch all other cases. Seems simpler to me this way. The comment needs an update, though, yes.

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td:2450]: I think I'd prefer to see this spelled out with 0b setting every bit explicitly

Comment (kosarev) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td]: Done.

**Review by arsenm — APPROVED** [review]

**Pull request merged**